runhouse/resources/hardware/cluster.py

```diff
@@ -65,6 +65,7 @@ def __init__(
         den_auth: bool = False,
         use_local_telemetry: bool = False,
         dryrun=False,
+        api_server_url=None,
```

Comment:
I think we should avoid having this in the constructor / factory if we can - this isn't really a user facing parameter